libr/bin/format/mach0/mach0.c

```diff
@@ -3311,92 +3321,89 @@ RSkipList *MACH0_(get_relocs)(struct MACH0_(obj_t) *bin) {
-				while (*ur.p++ && ur.p<end) {
-					/* empty loop */
 				}
-				if (threaded_binds) {
+				case BIND_OPCODE_SET_DYLIB_ORDINAL_IMM:
+					lib_ord = imm;
 					break;
-				}
-				sym_ord = -1;
-				if (bin->symtab && bin->dysymtab.nundefsym < 0xffff) {
-					for (j = 0; j < bin->dysymtab.nundefsym; j++) {
-						int stridx = 0;
-						int iundefsym = bin->dysymtab.iundefsym;
-						if (iundefsym >= 0 && iundefsym < bin->nsymtab) {
-							int sidx = iundefsym + j;
-							if (sidx < 0 || sidx >= bin->nsymtab) {
-								continue;
+				case BIND_OPCODE_SET_DYLIB_ORDINAL_ULEB:
+					lib_ord = ULEB();
+					break;
+				case BIND_OPCODE_SET_DYLIB_SPECIAL_IMM:
+					lib_ord = imm? (st8)(BIND_OPCODE_MASK | imm) : 0;
+					break;
+				case BIND_OPCODE_SET_SYMBOL_TRAILING_FLAGS_IMM: {
+					sym_name = (char*)ur.p;
+					while (*ur.p++ && ur.p<end) {
+						/* empty loop */
+					}
+					if (threaded_binds) {
+						break;
+					}
+					sym_ord = -1;
+					if (bin->symtab && bin->dysymtab.nundefsym < 0xffff) {
+						for (j = 0; j < bin->dysymtab.nundefsym; j++) {
+							int stridx = 0;
+							int iundefsym = bin->dysymtab.iundefsym;
+							if (iundefsym >= 0 && iundefsym < bin->nsymtab) {
+								int sidx = iundefsym + j;
+								if (sidx < 0 || sidx >= bin->nsymtab) {
+									continue;
+								}
+								stridx = bin->symtab[sidx].n_strx;
+								if (stridx < 0 || stridx >= bin->symstrlen) {
+									continue;
+								}
 							}
-							stridx = bin->symtab[sidx].n_strx;
-							if (stridx < 0 || stridx >= bin->symstrlen) {
-								continue;
+							if (!strcmp ((char *)bin->symstr + stridx, sym_name)) {
+								sym_ord = j;
+								break;
 							}
 						}
-						if (!strcmp ((char *)bin->symstr + stridx, sym_name)) {
-							sym_ord = j;
-							break;
-						}
 					}
+					break;
 				}
-				break;
-			}
-			case BIND_OPCODE_SET_TYPE_IMM:
-				type = imm;
-				break;
-			case BIND_OPCODE_SET_ADDEND_SLEB:
-				addend = SLEB();
-				break;
-			case BIND_OPCODE_SET_SEGMENT_AND_OFFSET_ULEB:
-				seg_idx = imm;
-				if (seg_idx < 0 || seg_idx >= bin->nsegs) {
-					bprintf ("Error: BIND_OPCODE_SET_SEGMENT_AND_OFFSET_ULEB"
-						" has unexistent segment %d\n", seg_idx);
-					R_FREE (opcodes);
-					r_skiplist_free (relocs);
-					r_pvector_free (threaded_binds);
-					return NULL; // early exit to avoid future mayhem
-				} else {
-					addr = bin->segs[seg_idx].vmaddr + ULEB();
-					segmentAddress = bin->segs[seg_idx].vmaddr \
-							+ bin->segs[seg_idx].vmsize;
-				}
-				break;
-			case BIND_OPCODE_ADD_ADDR_ULEB:
-				addr += ULEB();
-				break;
+				case BIND_OPCODE_SET_TYPE_IMM:
+					type = imm;
+					break;
+				case BIND_OPCODE_SET_ADDEND_SLEB:
+					addend = SLEB();
+					break;
+				case BIND_OPCODE_SET_SEGMENT_AND_OFFSET_ULEB:
+					seg_idx = imm;
+					if (seg_idx >= bin->nsegs) {
+						bprintf ("Error: BIND_OPCODE_SET_SEGMENT_AND_OFFSET_ULEB"
+							" has unexistent segment %d\n", seg_idx);
+						R_FREE (opcodes);
```

Comment:
just use free, because there's a return nearby and there's no need to nullify the pointer